File: qasdk/src/main/java/co/astrnt/qasdk/utils/ServiceUtils.java
package co.astrnt.qasdk.utils;

import android.app.ActivityManager;
import android.content.Context;

public class ServiceUtils {

    public static boolean isMyServiceRunning(Context context, Class<?> serviceClass) {
        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        assert manager != null;
        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
            if (serviceClass.getName().equals(service.service.getClassName())) {
                return true;
            }
        }
        return false;
    }
}

File: qasdk/src/main/java/co/astrnt/qasdk/utils/ServiceUtils.kt
package co.astrnt.qasdk.utils

import android.app.ActivityManager
import android.content.Context

object ServiceUtils {
    @JvmStatic
    fun isMyServiceRunning(context: Context, serviceClass: Class<*>): Boolean {
        val manager = (context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager)
        for (service in manager.getRunningServices(Int.MAX_VALUE)) {
            if (serviceClass.name == service.service.className) {
                return true
            }
        }
        return false
    }
}